src/gluonts/torch/model/predictor.py

```diff
@@ -73,10 +74,11 @@ def predict(
         self, dataset: Dataset, num_samples: Optional[int] = None
     ) -> Iterator[Forecast]:
         inference_data_loader = InferenceDataLoader(
-            dataset,
+            dataset=dataset,
             transform=self.input_transform,
```

Comment:
You could also just add the `SelectFields` transformation here maybe? This way `loader.py` stays untouched

Reply:
I am wondering if we should actually have select fields as an optional in every loader? I think you would face a similar problem for `ValidationDataLoader`.

Reply:
For now, I would go for as minimal a change as possible. It seems like (some) models that use `ValidationDataLoader` explicitly add `SelectField` https://github.com/lostella/gluonts/blob/c6f90a6afeea50fdc700ee8a0f7104c8d9f683da/src/gluonts/mx/model/deepar/_estimator.py#L419-L420